### PR TITLE
update clientInterface. update tsconfig.json to work with atom-typescript

### PR DESCRIPTION
**Pull request by saddieeiddas**
updated `clientInterface` with some missing properties.

updated `tsconfig.json` to work well with `atom-typescript`

updated version to `0.1.7`

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
+        "sourceMap": false,
+        "jsx": "react"
+    },
+    "compileOnSave": false,
+    "buildOnSave": false,
+    "filesGlob": [
+        "./src/**/*.ts",
+        "./src/**/*.tsx"
+    ],
+    "files": [
```
Comment (codecorsair) [inline tsconfig.json]: I really do not want to have to continually update this list of files.  Is this the only way it works with atom-typescript?

Comment (saddieeiddas) [inline tsconfig.json]: The list of files does not affect the `gulp build` process from my testing (if you have empty list of files or files missing the gulp build still builds everything), it is just something that atom updates automatically.

Comment (saddieeiddas) [inline tsconfig.json]: @codecorsair I have done more testing on this, and if you *dont* update the file it makes no difference to normal building. The gulp system just ignores the `files` parameter from `tsconfig`.

Comment (codecorsair) [inline tsconfig.json]: Ok, sounds good then

**Comment by codecorsair** [comment]
go for it!

**Comment by saddieeiddas** [comment]
@codecorsair I have updated this to fix conflicts and rebased everything. Should hopefully be ready to merge.

**Pull request merged**